TestAssignmentDoItSoftware/TestAssignmentDoItSoftware/TASingleton/TASingleton.m

```diff
@@ -0,0 +1,596 @@
+        case 3:{
+            
+            sortingPredicate = [[NSSortDescriptor alloc] initWithKey:@"priority" ascending:NO];
+            break;
+        }
+        case 4:{
+            
+            sortingPredicate = [[NSSortDescriptor alloc] initWithKey:@"dateto" ascending:YES];
+            break;
+        }
+        case 5:{
+            
+            sortingPredicate = [[NSSortDescriptor alloc] initWithKey:@"dateto" ascending:NO];
+            break;
+        }
+
+    }
+
+    NSArray *arrSortOrder = [[NSArray alloc] initWithObjects:sortingPredicate, nil];
+    fetchRequest.sortDescriptors = arrSortOrder;
+
+    
+    NSError *requestError = nil;
+    
+    [fetchRequest setIncludesPendingChanges:YES];
+    [fetchRequest setEntity:entity];
+    
+    NSArray *items = [self.managedObjectContext executeFetchRequest:fetchRequest error:&requestError];
+    
+    NSLog(@"records in Tasks: %ld", (long) [items count]);
+
+    for (task in items) {
+
+        NSArray *keys = [[[task entity] attributesByName] allKeys];
+        NSMutableDictionary *fetchedTask = [NSMutableDictionary dictionaryWithDictionary:[task dictionaryWithValuesForKeys:keys]];
+       
+        [fetchedTasks addObject:fetchedTask];
+        
+    }
+    
+    resultArray = [[NSArray alloc] initWithArray:fetchedTasks];
+    
+    return resultArray;
+}
+
+
+-(NSArray *)getNotifications{
+    
+    NSArray *resultArray;
+    NSMutableArray *fetchedTasks = [[NSMutableArray alloc] init];
+    Tasks *task;
+    
+    NSFetchRequest *fetchRequest = [[NSFetchRequest alloc] init];
+    NSEntityDescription *entity = [NSEntityDescription entityForName:@"Tasks" inManagedObjectContext:self.managedObjectContext];
+    
+    NSNumber *nid = [NSNumber numberWithUnsignedInteger:0];
+    
+    NSPredicate *predicateLid = [NSPredicate predicateWithFormat:@"nid > %@", nid];
+    
+    NSPredicate *compPredicate = [NSCompoundPredicate andPredicateWithSubpredicates:[NSArray arrayWithObjects:predicateLid, nil]];
+    [fetchRequest setPredicate:compPredicate];
+    
+    NSError *requestError = nil;
+    
+    [fetchRequest setIncludesPendingChanges:YES];
+    [fetchRequest setEntity:entity];
+    
+    NSArray *items = [self.managedObjectContext executeFetchRequest:fetchRequest error:&requestError];
+    
+    NSLog(@"records with notifications in Tasks: %ld", (long) [items count]);
+    
+    for (task in items) {
+        
+        NSArray *keys = [[[task entity] attributesByName] allKeys];
+        NSMutableDictionary *fetchedTask = [NSMutableDictionary dictionaryWithDictionary:[task dictionaryWithValuesForKeys:keys]];
+        
+        [fetchedTasks addObject:fetchedTask];
+        
+    }
+    
+    resultArray = [[NSArray alloc] initWithArray:fetchedTasks];
+    
+    return resultArray;
+}
+
+#pragma mark - Network interactions
+
+
+-(BOOL)isConnected{
+    
+    BOOL connected;
+    
+    connected = NO;
+    
+    NSURL *scriptUrl = [NSURL URLWithString:@"http://google.com"];
```

Comment:
Never assume `google.com` is going to be available. Use either your backend or the general iOS connectivity check.